core/src/impl/Kokkos_Error.cpp

```diff
@@ -102,11 +102,13 @@ void Experimental::RawMemoryAllocationFailure::print_error_message(
   o << "  (The allocation mechanism was ";
   switch (m_mechanism) {
     case AllocationMechanism::StdMalloc: o << "standard malloc()."; break;
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
     case AllocationMechanism::PosixMemAlign: o << "posix_memalign()."; break;
     case AllocationMechanism::PosixMMap: o << "POSIX mmap()."; break;
     case AllocationMechanism::IntelMMAlloc:
       o << "the Intel _mm_malloc() intrinsic.";
       break;
+#endif
```

Comment:
Does that make sense to preserve this when we just ignore the alloc mechanism while constructing the host memory space?

Reply:
`RawMemoryAllocationFailure` is defined in the `Experimental::` namespace so it is fair game for us to change.  Considering the warning I think we should get rid of it.

Reply:
Also preserving the error message for an allocation mechanism that we don't honor anyway makes little sense.

Reply:
fa90d1cfaaa91121b2ef1f2588296a1512d532a2